services/slam/client_test.go

```diff
@@ -275,6 +330,32 @@ func TestClientFailingService(t *testing.T) {
 		test.That(t, err.Error(), test.ShouldContainSubstring, "failure to get internal state")
 		test.That(t, internalState, test.ShouldBeNil)
 
+		// test get internal state stream
+		internalStateCallback, err := failingSLAMClient.GetInternalStateStream(context.Background(), nameFail)
+		test.That(t, err, test.ShouldBeNil)
+		fullBytes, err := helperConcatenateChunksToFull(internalStateCallback)
+		test.That(t, err.Error(), test.ShouldContainSubstring, "failure during get internal state stream")
+		test.That(t, fullBytes, test.ShouldBeNil)
+
 		test.That(t, conn.Close(), test.ShouldBeNil)
 	})
 }
+
+// Might be replaced by a non-test helper once GetPointCloudMapFull and GetInternalStateFull are created.
+func helperConcatenateChunksToFull(f func() ([]byte, error)) ([]byte, error) {
+	var fullBytes []byte
+	for {
+		chunk, err := f()
+		fmt.Println("yo chunk", chunk)
+		if err != nil {
+			return nil, err
+		}
+
+		fullBytes = append(fullBytes, chunk...)
+		fmt.Println("yo full bytes", fullBytes)
+		if len(chunk) < chunkSizeInternalState {
```

Comment:
I don't think this is correct. We should only stop once we get the EOF error. The size of the chunks shouldn't be the signal that we use to determine when to stop consuming from the callback function.